### ISSUE TITLE
Should Status contain all fields, or only fields which are not already in Spec?

### ISSUE THREAD

**Issue by matthchr** [Azure/k8s-infra#269]
It's a slightly foreign pattern to duplicate information like we're currently doing in Kubernetes, but the disadvantage of not duplicating is that there's no way for customers to realize that their resource doesn't match the spec without going outside of the context of the operator.

**Comment by matthchr** [comment]
For what it's worth, Crossplane doesn't duplicate the data currently.

**Comment by babbageclunk** [comment]
I had assumed that we'd want to keep the spec fields a subset of the status ones because that provides a reasonable way to check whether the actual state matches the desired state. I guess we could have that information in the values coming from ARM but drop the spec-covered fields from the status we write out to the K8s API. 

That said, it seems like users would want to see what the actual value for some of those overlapping fields are in status to understand why some resource isn't converging. Possibly that's thinking too much like the developer of the service and not enough like the user though? We could provide that information in a different form that didn't require the user to diff spec and status by eye.

**Comment by matthchr** [comment]
yeah we could have an annotation like "is synced" or something, that does the diff and collapses it into a boolean for people.

With that said I think the primary argument against this is "most Kubernetes resources don't do it". Without that, I think duplicating info in the spec and status. So the question for me at least really boils down to how much we want to match standard Kubernetes practices vs how much we think our use case differs enough to do it a bit differently.

I doubt duplicating into status would really cause customers a lot of confusion?

**Comment by Porges** [comment]
The way that the [API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status) document discusses status makes it seem like it _should_ be a superset:
> Over time the system will work to bring the "status" into line with the "spec".
